marklogic-data-hub-central/ui/src/components/queries/saving/save-query-modal/save-query-modal.tsx

```diff
@@ -0,0 +1,154 @@
+import React, {useState, useContext} from 'react';
+import {Modal, Form, Input, Radio, Button} from 'antd';
+import {SearchContext} from "../../../../util/search-context";
+import styles from './save-query-modal.module.scss';
+
+interface Props {
+    setSaveModalVisibility: () => void;
+    saveNewQuery: (queryName: string, queryDescription: string,facets:{}) => void;
+    greyFacets:any[];
+    toggleApply: (clicked:boolean) => void;
+    toggleApplyClicked: (clicked:boolean) => void;
+}
+
+
+const SaveQueryModal: React.FC<Props> = (props) => {
+
+    const {
+        clearAllGreyFacets,
+        greyedOptions,
+        setAllSearchFacets,
+        searchOptions,
+    } = useContext(SearchContext);
+
+
+    const [queryName, setQueryName] = useState('');
+    const [queryDescription, setQueryDescription] = useState('');
+    const [radioOptionClicked, setRadioOptionClicked] = useState(0);
+    const [queryEmpty, isQueryEmpty] = useState<any>('');
+
+    const layout = {
+        labelCol: { span: 6 },
+        wrapperCol: { span: 18 },
+    };
+
+    const onCancel = () => {
+        props.setSaveModalVisibility();
+    }
+
+    const onOk = () => {
+        let facets = {...searchOptions.selectedFacets}
+        switch(radioOptionClicked) {
+            case 1:
+                setAllSearchFacets(searchOptions.selectedFacets);
+                setAllSearchFacets(greyedOptions.selectedFacets);
+                facets = {...facets,...greyedOptions.selectedFacets};
+                clearAllGreyFacets();
+                props.toggleApplyClicked(true);
+                props.toggleApply(false);
+                break;
+            case 2:
+                break;
+            case 3:
+                clearAllGreyFacets();
+                props.toggleApplyClicked(true);
+                props.toggleApply(false);
+        }
+
+        if(queryName.length > 0 ){
+            props.saveNewQuery(queryName, queryDescription, facets);
+        } else {
+            isQueryEmpty('error')
+        }
+    }
+
+    const handleChange = (event) => {
+        if (event.target.id === 'save-query-name') {
+           setQueryName(event.target.value);
```

Comment:
I think there needs to be some basic validation for the name field. If save a query as an empty string, it doesn't show up in the save query dropdown, but it shows in the manage queries table.

Reply:
It shows on the save query dropdown and manage query for me. 
<img width="255" alt="Screen Shot 2020-04-27 at 8 46 44 AM" src="https://user-images.githubusercontent.com/44847151/80392670-4db47280-8864-11ea-807d-1730702b19d6.png">

Reply:
@brucean52 , @timur-isangulov, added the fixes.